### PR TITLE
adding provider ID to op and verify error before failing the op

### PR DESCRIPTION
**Pull request by WendelHime**

### CODE REVIEW

File: fronted.go [hunk 1]
```diff
@@ -386,12 +390,31 @@ func (f *fronted) doDial(m MasqueradeInterface) (conn net.Conn, retriable bool,
...
+		}
+
+		// Fallback to message-based checking (works for Windows and Unix-like systems)
+		errMsg := opErr.Err.Error()
+		if strings.Contains(errMsg, "network is unreachable") ||
+			strings.Contains(errMsg, "no route to host") ||
+			strings.Contains(errMsg, "unreachable network") ||
+			strings.Contains(errMsg, "unreachable host") {
+			return true
+		}
```
Comment (garmr-ulfr) [inline fronted.go]: Which OSes would this not work for? MacOS, iOS, and Android are all Unix based. AFAIK, this _should_ work with all platforms we support.

**Review by garmr-ulfr — APPROVED** [review]
Just one comment for clarification but, otherwise LGTM!

File: fronted.go (hunk 1 above)
Comment (WendelHime) [reply fronted.go]: Those string validation should cover everything. I guess the errors check above cover Linux/Unix and this string errors should cover the windows

Comment (WendelHime) [reply fronted.go]: Looking at similar errors at honeycomb you should be able to see errors grouped by OS

Comment (garmr-ulfr) [reply fronted.go]: > I guess the errors check above cover Linux/Unix and this string errors should cover the windows

Oh OK. Would you add that to the comment to make it more clear?

**Pull request merged**